libs/pilight/protocols/433.92/xiron.c

```diff
@@ -0,0 +1,211 @@
+/*
+	Copyright (C) 2019 CurlyMo & kithack
+
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "../../core/pilight.h"
+#include "../../core/common.h"
+#include "../../core/dso.h"
+#include "../../core/log.h"
+#include "../protocol.h"
+#include "../../core/binary.h"
+#include "../../core/gc.h"
+#include "xiron.h"
+
+#define PULSE_MULTIPLIER	20
+#define MIN_PULSE_LENGTH	94
+#define MAX_PULSE_LENGTH	141
+#define AVG_PULSE_LENGTH	151
+#define RAW_LENGTH		74
+
+typedef struct settings_t {
+	double id;
```

Comment:
Why are these doubles and not just integers?

Reply:
All other 433 MHz protocols use doubles there. So I didn't change that.

Reply:
Could be that they have been wrong for all this time :wink:

Reply:
True. It doesn't make sense

Reply:
Fixed it